esrally/mechanic/telemetry.py

```diff
@@ -481,107 +481,90 @@ def record(self):
+            collected_node_stats = {}
+
             if self.include_indices:
-                self.record_indices_stats(node_name, node_stats,
-                                          include=["docs", "store", "indexing", "search", "merges", "query_cache", "fielddata",
-                                                   "segments", "translog", "request_cache"])
+                collected_node_stats.update(
+                    self.collect_indices_stats(node_name, node_stats,
+                                               include=["docs", "store", "indexing", "search", "merges", "query_cache", "fielddata",
+                                                        "segments", "translog", "request_cache"]))
             if self.include_thread_pools:
-                self.record_thread_pool_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_thread_pool_stats(node_name, node_stats))
             if self.include_breakers:
-                self.record_circuit_breaker_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_circuit_breaker_stats(node_name, node_stats))
             if self.include_buffer_pools:
-                self.record_jvm_buffer_pool_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_jvm_buffer_pool_stats(node_name, node_stats))
             if self.include_mem_stats:
-                self.record_jvm_mem_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_jvm_mem_stats(node_name, node_stats))
             if self.include_network:
-                self.record_network_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_network_stats(node_name, node_stats))
             if self.include_process:
-                self.record_process_stats(node_name, node_stats)
+                collected_node_stats.update(self.collect_process_stats(node_name, node_stats))
+
+            self.metrics_store.put_doc(collected_node_stats,
+                                       level=MetaInfoScope.node,
+                                       node_name=node_name,
+                                       meta_data=self.metrics_store_meta_data)
 
         time.sleep(self.sample_interval)
 
-    def record_indices_stats(self, node_name, node_stats, include):
+    def flatten_stats_fields(self, prefix=None, stats=None, include=None, level=0):
+        """
+        Flatten provided dict using an optional prefix and top level key filters.
+
+        :param prefix: The prefix for all flattened values. _ will be used as a separator. Defaults to None.
+        :param stats: Dict with values to be flattened, using _ as a separator. Defaults to {}.
+        :param include: List of keys to process in stats. Filtering will be done only for top level keys.
+        :param level: Int used to control filtering at the top level; only used internally. Defaults to 0.
```

Comment:
We could have an internal function that defines this parameter and hide it from the top-level function?